docs/tutorials/demo_best_practices.py

```diff
@@ -0,0 +1,271 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_markers: '"""'
+#     cell_metadata_filter: -all
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.15.2
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+"""
+# Data Versioning Best Practices
+
+This notebook will guide you through the best practices for versioning data in a data science project.
+We assume you have lakeFS and lakeFS-spec setup. If you need help with it, look into the other parts of the docs or the other tutorial which is more focused on the setup.
+
+We will explain the following best practices for data versioning in this example:
+- Define a data repository
+- Follow a branching strategy that ensures data integrity on the `main` branch, e.g. by running tests on feature branch datasets.
+- Utilize reusable and tested functions for data transformation.
+- Use commits to save checkpoints and merge branches for atomic changes.
+- Keep naming (of branches and commits) consistent, concise, and unique.
+- Use descriptive naming where it matters .
+
+For this demo project, we aim to build a weather predictor using data from a public API.
+This simulates the dynamics within a real world scenario where we continuously collect more data.
```

Comment:
Unfortunate wording, as "simulates the dynamics" sounds more like what the model does instead of the demo. I would go for something like "This is a less complex example of a real-world scenario [...]".